### PR TITLE
Bygex supported integration of space transition config into map.dms

### PR DESCRIPTION
**Pull request by Jordie0608**
#11643

This replaces the space_levels.txt config of space transitions by integrating it into the existing map define files to avoid the issues outlined in https://github.com/tgstation/-tg-station/pull/11643#issuecomment-139133978

I believe I've converted the vars in /proc/setup_map_transitions() correctly however I'd suggest not merging this until @Razharas has confirmed that.

### CODE REVIEW

**Comment by Mloc** [comment]
no.
- won't work for people who run servers in a different dir to where they're compiled
- won't work for people who don't use bygex

**Comment by Jordie0608** [comment]
1. I'm not entirely sure how that'd work but it's easily addressed with a variable path set by the host defaulting to null
2. Tg code already requires bygex

**Comment by optimumtact** [comment]
also +1 on bygex, bygex has been a hard requirement for a while now, which is why I am maintain a linux version.

**Comment by Razharas** [comment]
>Comments are defines of map setups

http://www.youtube.com/watch?v=UZuzw2TCNSU

**Comment by Anonus** [comment]
>Comments being important

Uh, that's really weird

File: code/modules/space transition/space_transition.dm [hunk 1]
```diff
@@ -74,48 +74,47 @@ var/list/z_levels_list = list()
...
-	var/list/config_settings[SLS.len][]
-	for(var/A in SLS)
-		config_settings[SLS.Find(A)] = text2list(A, ";")
-	var/conf_set_len = SLS.len
-	SLS.Cut()
-	for(var/A in config_settings)
+	var/datum/regex/initresults = regex_find(configtext, "zlevel amount:(\\d+)")
+	var/zlen = initresults.str(2)
+	for(var/i, i>zlen, i++)
+		var/datum/regex/results = regex_find(configtext, "(\\w+)\\.dmm\"\\/\\/([i]) (unlinked|linked)")
```
Comment (Razharas) [inline code/modules/space transition/space_transition.dm]: What prevernt this code from finding same first line every time?
Do you remove the resulting text from configtext somewhere?

Comment (Jordie0608) [inline code/modules/space transition/space_transition.dm]: The value of i is part of the regex, which is incremented each pass of the loop.

**Comment by Mloc** [comment]
instead of this please use an assoc in the map def file, it's -much- cleaner, more consistent, and completely avoids the source dir synchronization issue.

**Comment by Razharas** [comment]
After some talks in coderbus i reached the conclusion that having comments decide if the map is linked is horrible
correct way is to define same list var in each dm (they are all mutually exclusive) that holds the map states

**Comment by Jordie0608** [comment]
I'm closing this for, as discussed in coderbus, I disagree with the reasons slated against this code, it's not an essential change as Raz's system does function as intended and because I've moved onto other stuff and can't be bothered coming back to this yet.

**Pull request closed without merging**